lib/src/rules/avoid_relative_imports.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../ast.dart';
+
+const _desc = r'Avoid relative imports for all files`.';
+
+const _details = r'''*DO* avoid relative imports for files in `lib/`.
+
+When mixing relative and absolute imports it's possible to create confusion
+where the same member gets imported in two different ways. One way to avoid
+that is to ensure you consistently use absolute imports for files withing the
+`lib/` directory.
+
+This is the opposite of 'prefer_relative_imports'.
+
+**GOOD:**
+
+```
+import 'package:foo/bar.dart';
+
+import 'package:foo/baz.dart';
+
+import 'package:foo/src/baz.dart';
+...
+```
+
+**BAD:**
+
+```
+import 'baz.dart';
+
+import 'src/bag.dart'
+
+import '../lib/baz.dart';
+
+...
+```
+
+''';
+
+class AvoidRelativeImports extends LintRule implements NodeLintRule {
+  AvoidRelativeImports()
+      : super(
+            name: 'avoid_relative_imports',
```

Comment:
You had no way of knowing, but we have decided that it was a mistake to use "avoid", "prefer" and other terms from the style guide in the names of lints. Perhaps we could name this something like "always_use_package_imports" (I'm open to alternatives).

Reply:
renamed to `always_use_package_imports`